runtime/wasmer.go

```diff
@@ -407,7 +470,25 @@ func NewRuntime(fp string, t *trie.Trie) (*Runtime, error) {
 		return nil, err
 	}
 
-	data := unsafe.Pointer(t)
+	memAllocator := newAllocator(&instance.Memory, 0)
+
+	runtimeCtx := &RuntimeCtx{
+		trie:      t,
+		allocator: &memAllocator,
+	}
+	// add runtimeCtx to registry
+	mutex.Lock()
```

Comment:
very cool solution, just wondering why we need a mutex?

Reply:
I think a mutex is needed to protect registry map from possible concurrent access.  I added more comments to the code.